src/plugins/janus_videoroom.c

```diff
@@ -6865,6 +6865,18 @@ void janus_videoroom_setup_media(janus_plugin_session *handle) {
 			janus_videoroom_publisher *participant = janus_videoroom_session_get_publisher(session);
 			/* Notify all other participants that there's a new boy in town */
 			janus_videoroom_notify_about_publisher(participant, FALSE);
+			/* Check if we need to start recording */
+			janus_mutex_lock(&participant->rec_mutex);
+			if(participant->room && participant->room->record) {
```

Comment:
I guess there should be an OR with `participant->recording_active` here, since we have a similar check in the other places where we auomatically start recording (like the already mentioned post-answer check).

Reply:
ah, i seem to have misunderstood the meaning of `participant->recording_active`, thanks.